src/solver/solver.cxx

```diff
@@ -472,8 +473,21 @@ int Solver::solve(int NOUT, BoutReal TIMESTEP) {
     throw BoutException(_("Failed to initialise solver-> Aborting\n"));
   }
 
+  // Set the run ID
+  run_restart_from = run_id; // Restarting from the previous run ID
+  run_id = createRunID();
+
+  // Put the run ID into the options tree
+  // Forcing in case the value has been previously set
+  Options::root()["run"]["run_id"].force(run_id, "Solver");
+  Options::root()["run"]["run_restart_from"].force(run_restart_from, "Solver");
+
   /// Run the solver
   output_info.write(_("Running simulation\n\n"));
+  output_info.write("Run ID: {:s}\n", run_id);
+  if (run_restart_from != "zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz") {
```

Comment:
```suggestion
  if (run_restart_from != default_run_id) {
```

Reply:
Good catch, thanks!

Thinking about this some more, is there a reason why we couldn't just make the default the empty string, then check for `run_{id,restart_from}.empty()`? Does it interact funny when writing to disk?

Reply:
I think having 36-character strings is sometimes necessary. The Datafile string I/O assumes that the length of the string doesn't change between adding the variable to the Datafile object, and reading/writing. I think that means `run_id` and `run_restart_from` need to be 36-character strings by default. Possibly we could get away with `run_restart_from` being an empty string, but I think it's less confusing and potentially safer to have both be 36-character strings.

Reply:
Ok, we may be able to change this when we do away with Datafile